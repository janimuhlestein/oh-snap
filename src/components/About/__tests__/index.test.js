import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import About from '..';

const categories = [
    { name: 'portraits', description: 'Portraits of people in my life' }
];
const mockCurrentCategory = jest.fn();
const mockSetCurrentCategory = jest.fn();

//cleanup after each test
afterEach(cleanup);

describe('About component', ()=> {
    //renders About test
    //first test
    it('renders', () => {
        render(<About 
            categories = {categories}
            setCurrentCategory = {mockSetCurrentCategory}
            currentCategory={mockCurrentCategory}
        />);
    });

    //second test
    it('matches snapshot DOM node structure', () => {
        const { asFragment } = render(<About />);
        expect(asFragment()).toMatchSnapshot();
    })
});