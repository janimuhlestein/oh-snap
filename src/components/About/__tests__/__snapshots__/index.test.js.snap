// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section
    class="my-5"
  >
    <h1
      id="about"
    >
      Who am I
    </h1>
    <img
      alt="cover"
      class="my-3"
      src="cover-image.jpg"
      style="width: 100%;"
    />
    <div
      class="my-2"
    >
      <p>
        I am a professional photographer, specializing in product images.
      </p>
    </div>
  </section>
</DocumentFragment>
`;
